### PR TITLE
feat: Add the option `ignored-patterns` to the rule `format-comment`

### PR DESCRIPTION
**Pull request by ValentinVignal**
### What is the purpose of this pull request? (put an "X" next to an item)

[ ] Documentation update
[ ] Bug fix
[ ] New rule
[x] Changes an existing rule
[ ] Add autofixing to a rule
[ ] Add a CLI option
[ ] Add something to the core
[ ] Other, please explain:

Should close https://github.com/dart-code-checker/dart-code-metrics/issues/826

### What changes did you make? (Give an overview)

I added the option `ignored-patterns` (a list of strings that will be used as regexes) to the rule `format-comment` to be able to ignore some kind of comments.

### Is there anything you'd like reviewers to focus on?

Am I correctly getting the `ignored-patterns` option from the `config` ? I didn't find how to test it myself (e2e test) and make sure my code is actually working.

Also, do I have to apply any change to your web site? If yes, how?

### CODE REVIEW

File: lib/src/analyzers/lint_analyzer/rules/rules_list/format_comment/format_comment_rule.dart [hunk 1]
```diff
@@ -24,15 +24,22 @@ class FormatCommentRule extends CommonRule {
   static const _warning = 'Prefer formatting comments like sentences.';
 
   FormatCommentRule([Map<String, Object> config = const {}])
-      : super(
```
Comment (incendial) [inline lib/src/analyzers/lint_analyzer/rules/rules_list/format_comment/format_comment_rule.dart:27]: The idea is right, but we usually create a _ConfigParser class to incapsulate the parsing. Could you create it here too? You can find an example here  https://github.com/dart-code-checker/dart-code-metrics/blob/master/lib/src/analyzers/lint_analyzer/rules/rules_list/avoid_returning_widgets/config_parser.dart

File: lib/src/analyzers/lint_analyzer/rules/rules_list/format_comment/visitor.dart [hunk 2]
```diff
@@ -11,6 +11,10 @@ const _ignoreExp = 'ignore:';
 const _ignoreForFileExp = 'ignore_for_file:';
 
 class _Visitor extends RecursiveAstVisitor<void> {
+  _Visitor(this._ignoredPatterns);
+
+  final Iterable<RegExp> _ignoredPatterns;
```
Comment (incendial) [inline lib/src/analyzers/lint_analyzer/rules/rules_list/format_comment/visitor.dart:16]: And please reorder variables as here https://github.com/dart-code-checker/dart-code-metrics/blob/master/lib/src/analyzers/lint_analyzer/rules/rules_list/avoid_returning_widgets/visitor.dart#L15

File: lib/src/analyzers/lint_analyzer/rules/rules_list/format_comment/format_comment_rule.dart [hunk 3]
```diff
@@ -24,15 +24,22 @@ class FormatCommentRule extends CommonRule {
...
+        ).map((stringPattern) => RegExp(stringPattern)),
+        super(
           id: ruleId,
           severity: readSeverity(config, Severity.style),
           excludes: readExcludes(config),
         );
 
+  /// The patterns to ignore. They are used to ignore and not lint comments that
+  /// match at least one of them.
+  final Iterable<RegExp> _ignoredPatterns;
```
Comment (incendial) [inline lib/src/analyzers/lint_analyzer/rules/rules_list/format_comment/format_comment_rule.dart:38]: Please declare this variable before the constructor.

**Review by incendial — COMMENTED** [review]
Thank you for adding this option! 

> Am I correctly getting the ignored-patterns option from the config

Yes, every thing is fine, but I left few small comments just to keep the codebase in the same code style. Could you address them?

**Comment by incendial** [comment]
To update the website, you need to change this file https://github.com/dart-code-checker/dart-code-metrics/blob/master/website/docs/rules/common/format-comment.md. Here is an example https://github.com/dart-code-checker/dart-code-metrics/blob/master/website/docs/rules/common/no-equal-arguments.md

Please also add the `configurable` label to this rule in https://github.com/dart-code-checker/dart-code-metrics/blob/master/website/docs/rules/overview.md

File: lib/src/analyzers/lint_analyzer/rules/rules_list/format_comment/format_comment_rule.dart (hunk 1 above)
Comment (ValentinVignal) [reply lib/src/analyzers/lint_analyzer/rules/rules_list/format_comment/format_comment_rule.dart:27]: I created a config parser here https://github.com/dart-code-checker/dart-code-metrics/pull/828/commits/06b475e10834de19ad9fed37bbafc8ee677e0d31

File: lib/src/analyzers/lint_analyzer/rules/rules_list/format_comment/visitor.dart (hunk 2 above)
Comment (ValentinVignal) [reply lib/src/analyzers/lint_analyzer/rules/rules_list/format_comment/visitor.dart:14]: I moved the variables in this commit: https://github.com/dart-code-checker/dart-code-metrics/pull/828/commits/5ef888526b251c5dbacaf2e7c2689ab933ed06ad

File: lib/src/analyzers/lint_analyzer/rules/rules_list/format_comment/format_comment_rule.dart (hunk 3 above)
Comment (ValentinVignal) [reply lib/src/analyzers/lint_analyzer/rules/rules_list/format_comment/format_comment_rule.dart]: I moved the variables in this commit: https://github.com/dart-code-checker/dart-code-metrics/pull/828/commits/5ef888526b251c5dbacaf2e7c2689ab933ed06ad

**Comment by ValentinVignal** [comment]
> To update the website, you need to change this file https://github.com/dart-code-checker/dart-code-metrics/blob/master/website/docs/rules/common/format-comment.md. Here is an example https://github.com/dart-code-checker/dart-code-metrics/blob/master/website/docs/rules/common/no-equal-arguments.md
> 
> Please also add the `configurable` label to this rule in https://github.com/dart-code-checker/dart-code-metrics/blob/master/website/docs/rules/overview.md

I updated the documentation in this commit: https://github.com/dart-code-checker/dart-code-metrics/pull/828/commits/6288e6f806130d13ecb4a67685a68742185b8aeb


Tell me if this is too specific/not good enough

**Review by dkrutskikh — APPROVED** [review]

**Review by ewexs — APPROVED** [review]

**Pull request merged**